app/classifier/tasks/drawing/hide-previous-marks-toggle.cjsx

```diff
@@ -7,18 +7,17 @@ HidePreviousMarksToggle = React.createClass
     classification: null
     onChange: ->
 
-  componentDidMount: ->
-    @_lastKnownAnnotationCount = @props.classification.annotations.length
-
-  componentWillReceiveProps: (nextProps) ->
-    # Automatically un-hide previous marks.
-    if nextProps.classification.annotations.length isnt @_lastKnownAnnotationCount
-      @setPreviousMarks 0
+  getInitialState: ->
+    checked: false
 
   setPreviousMarks: (count) ->
-    @_lastKnownAnnotationCount = @props.classification.annotations.length
     @props.classification._hideMarksBefore = count
-    @props.onChange @props.classification
+    checked = count > 0
+    @setState { checked }
+    # this is a hack to force the parent to render again with marks hidden
```

Comment:
should this be extracted to a function with a name like `force_parent_render_hack()` to signify the intention of it? 

It might allow the removal of the comment and the commit history could store the context of the reason for that code?

Reply:
That's not a bad idea, though the linter would make us call it `forceParentRender` because it complains if names aren't camel-case.

Reply:
i can't help myself when writing code snippets these days, i should have 🐫'd it